### ISSUE TITLE
Incorrect routing for requests without leading slash

### ISSUE THREAD

**Issue by syzygymsu** [symfony/symfony#21967]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | all

HTTP requests without leading "/" (e.g. `GET ?a=b ...`) are not routed to index controller (one with path="/").

How to reproduce:
Create a new Symfony project and configure it to run under a root of any hostname (let it be `http://localhost/`).
Do not use a browser because majority of them do not produce requests like the one in question. Instead use telnet:
```
telnet localhost 80
> GET ?a=b HTTP/1.1
> Host: localhost
>
```
Expected output: welcome page (same as for `GET /?a=b ...`).
Actual output: "404 Not Found" page.

Reproduced with latest versions of v2.8 and v3.2 but other releases should be affected too.

**Comment by jakzal** [comment]
The http specification [defines Request-Uri](https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html#sec5.1.2) as:

```Request-URI    = "*" | absoluteURI | abs_path | authority```

The specification also says:

> The most common form of Request-URI is that used to identify a resource on an origin server or gateway. In this case the absolute path of the URI MUST be transmitted (see section 3.2.1, abs_path) as the Request-URI [..]

and:

> Note that the absolute path cannot be empty; if none is present in the original URI, it MUST be given as "/" (the server root).

In your case you're not sending the path at all, but only a query string (`GET ?a=b HTTP/1.1`),

**Comment by syzygymsu** [comment]
Yes, in theory such requests should not exist. But in practice I'm receiving them in production environment and Apache httpd server accepts them. And php built-in server is not for production anyway.

**Comment by jakzal** [comment]
That's why I said I'm not sure "how it is with real web servers". If apache lets such requests pass through, we need to handle them indeed.

**Comment by xabbuh** [comment]
Is the behaviour of the built-in web server correct according to the spec? If so, we could let the kernel respond with an "Invalid Request" response too when detecting such requests, can't we?

**Comment by jakzal** [comment]
@xabbuh I thought about it, but it would still leave the request object in an invalid state with a query string in the path info.

**Comment by javiereguiluz** [comment]
Can't we check if the path is empty and then initialize it to `/` ?

**Comment by xabbuh** [comment]
@jakzal You are right. I think this could be solved first by #21984.
